src/transactions/TransactionFrame.cpp

```diff
@@ -875,7 +875,8 @@ TransactionFrame::isTooEarlyForAccount(LedgerHeaderWrapper const& header,
 
 bool
 TransactionFrame::commonValidPreSeqNum(
-    AppConnector& app, LedgerSnapshot const& ls, bool chargeFee,
+    AppConnector& app, std::optional<SorobanNetworkConfig> cfg,
```

Comment:
Nit: Can we make  these `std::optional<SorobanNetworkConfig>& cfg` to avoid extra copies?